### PR TITLE
Setup haml-lint CI with todo config

### PR DESCRIPTION
**Pull request by nschonni**
Noticed the config file was already in the repo, but wasn't wired up anywhere.
- Adds the dependency
- Generate the todo file
- Extent the TODOs and remove all the config that shadowed the default config
- Add CI job that runs when the HAML files are touched or a dependency upgrade
- Added a problem matcher to comment on the PRs

### CODE REVIEW

**Comment by nschonni** [comment]
Sample annotation after enabling one of the rules
image

File: .haml-lint_todo.yml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      - 'app/views/statuses/_poll.html.haml'
+      - 'app/views/statuses/_status.html.haml'
+      - 'app/views/statuses/show.html.haml'
+      - 'app/views/user_mailer/suspicious_sign_in.html.haml'
+      - 'app/views/user_mailer/warning.html.haml'
+      - 'app/views/user_mailer/webauthn_credential_added.html.haml'
+      - 'app/views/user_mailer/webauthn_credential_deleted.html.haml'
+
+  # Offense count: 918
+  LineLength:
```
Comment (nschonni) [inline .haml-lint_todo.yml:56]: May be better to just move this from the TODO to a regular disabled rule if it's unlikely to be fixed

File: app/views/settings/applications/index.html.haml [hunk 2]
```diff
@@ -5,7 +5,7 @@
   = link_to t('doorkeeper.applications.index.new'), new_settings_application_path, class: 'button'
 
 - if @applications.empty?
-  %div.muted-hint.center-text=t 'doorkeeper.applications.index.empty'
+  .muted-hint.center-text= t('doorkeeper.applications.index.empty')
```
Comment (nschonni) [inline app/views/settings/applications/index.html.haml:8]: Minimize diff
```suggestion
  .muted-hint.center-text= t 'doorkeeper.applications.index.empty'
```

File: .github/workflows/lint-haml.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      - name: Clone repository
+        uses: actions/checkout@v3
+
+      - name: Install native Ruby dependencies
+        run: sudo apt-get install -y libicu-dev libidn11-dev
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
```
Comment (nschonni) [inline .github/workflows/lint-haml.yml:34]:
```suggestion
          ruby-version: .ruby-version
          bundler-cache: true
```

**Review by ykzts — APPROVED** [review]

**Comment by nschonni** [comment]
@Gargron do you prefer the cleanup split off from the main CI setup?

**Pull request merged**